### DISCUSSION TITLE
Change loadable.state to "loading", "success", and "error"

### DISCUSSION THREAD

**Discussion by hnordt** [pmndrs/jotai#2850] [Ideas] [answered]
loadable.state being “hasError” or “hasData” feels a bit off to me.

How about renaming it to something simpler like “error” or “success”? What do you think?

**Comment by dai-shi** [comment]
It's kind of too late. Migration isn't very easy and I don't know if it's worth it.

It's not that bad from the type definition. Maybe `.state` wasn't very appropriate.

https://github.com/pmndrs/jotai/blob/d2a42f785b9803764b97fbb0ccd349d2d99069e9/src/vanilla/utils/loadable.ts#L11-L14

**Reply by hnordt** [reply]
What about adding a new prop “status” and deprecating “state”?

type Status = “loading” | “loaded” | “errored”

Or

type Status = “loading” | “success” | “error”

**Reply by dai-shi** [reply, marked as answer]
That's technically possible. I would like to get more feedback on this renaming.

**Comment by wojtekmaj** [comment]
With loadable being deprecated, this discussion could likely be closed.
